### PR TITLE
[CPU] Fix RegionYolo (v3) node and Ngraph reference implementations

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### Details:
 - Destination size was calculated incorrectly for v3 versions

### Tickets:
 - CVS-56041

### CODE REVIEW

**Comment by EgorDuplensky** [comment]
RegionYolo [description](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/RegionYolo_1.md#regionyolo--openvino_docs_ops_detection_regionyolo_1)

**Review by dmitry-gorokhov — APPROVED** [review]
CPU part lgtm

**Comment by dmitry-gorokhov** [comment]
@lazarevevgeny could you please review ngraph reference changes? thanks

File: ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp [hunk 1]
```diff
@@ -86,24 +86,27 @@ namespace ngraph
...
                 }
                 else
                 {
                     // Yolo layer (Yolo v3)
                     num_regions = mask_size;
                     end_index = width * height * (classes + 1);
+                    output_size = width * height * num_regions * (classes + coords + 1);
                 }
 
+                std::copy(input, input + output_size, output);
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp:108]: Do we need to add more tests for nGraph reference implementation?

**Review by lazarevevgeny — APPROVED** [review]
The update in output size calculation is correct.

But I want to see more test to check that output values are calculated correctly.

File: ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp (hunk 1 above)
Comment (EgorDuplensky) [reply ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp:108]: Are tests for the reference implementation located in test/backend?
As for more tests, I could templatize them and basically replicate parameters from CPU tests, is it ok?

**Comment by EgorDuplensky** [comment]
Rebasing

File: ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp:108]: I propose to extend current backend tests in order to cover the fix in the nGraph unit-tests.

Comment (EgorDuplensky) [reply ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp:108]: Added test, which reproduce the issue (fails on master)

File: ngraph/test/backend/region_yolo.in.cpp [hunk 2]
```diff
@@ -70,3 +70,33 @@ NGRAPH_TEST(${BACKEND_NAME}, region_yolo_v3_mxnet)
...
+
+    const auto A = make_shared<op::Parameter>(element::f32, input_shape);
+    const auto R = make_shared<op::v0::RegionYolo>(A, coords, classes, num, false, mask, axis, end_axis);
+    const auto f = make_shared<Function>(R, ParameterVector{A});
+
+    EXPECT_EQ(R->get_output_shape(0), output_shape);
+
+    auto test_case = test::TestCase<TestEngine>(f);
+
+    test_case.add_input_from_file<float>(input_shape, TEST_FILES, "region_in_yolov3_mxnet_2.data");
```
Comment (lazarevevgeny) [inline ngraph/test/backend/region_yolo.in.cpp:98]: Do not add more binary files with test data! Let's generate random data or prepare a small test hardcoded as a plain text in the test itself.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/test/backend/region_yolo.in.cpp (hunk 2 above)
Comment (EgorDuplensky) [reply ngraph/test/backend/region_yolo.in.cpp]: Reimplemented

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**